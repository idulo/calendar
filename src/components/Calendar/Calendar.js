import React from "react";
import moment from "moment";

import MonthPicker from "../MonthPicker";
import CalendarDays from "../CalendarDays";
import ReminderForm from "../ReminderForm";
import ReminderEntry from "../ReminderEntry";
import STYLES from "./Calendar.module.css";

class Calendar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      date: moment(),
      showReminderWindow: false,
      selectedDay: 0,
      reminders: []
    };
  }

  currentMonthAndYear = () => moment(this.state.date).format("MMMM YYYY");

  goToPreviousMonth = () =>
    this.setState({
      date: moment(this.state.date).subtract(1, "months")
    });

  goToNextMonth = () =>
    this.setState({
      date: moment(this.state.date).add(1, "months")
    });

  toggleReminder = day =>
    this.setState({
      showReminderWindow: !this.state.showReminderWindow,
      selectedDay: day
    });

  addReminder = reminderText => {
    this.state.reminders.push({
      month: this.state.date.month(),
      day: this.state.selectedDay,
      text: reminderText
    });
    this.toggleReminder();

    console.log("Reminders list: ");
    console.log(this.state.reminders);
  };

  render() {
    return (
      <div className={STYLES.container}>
        <div className={STYLES.calendarContainer}>
          <MonthPicker
            goToNextMonth={this.goToNextMonth}
            goToPreviousMonth={this.goToPreviousMonth}
            month={this.currentMonthAndYear()}
          />
          <CalendarDays
            date={this.state.date}
            addReminder={this.addReminder}
            toggleReminder={this.toggleReminder}
          />
        </div>
        <div className={STYLES.sidepanel}>
          {this.state.showReminderWindow ? (
            <ReminderForm
              addReminder={this.addReminder}
              selectedDay={this.state.selectedDay}
              selectedMonth={this.state.date.month()}
            />
          ) : null}
          {this.state.reminders.map(reminder => (
            <ReminderEntry reminder={reminder} />
          ))}
        </div>
      </div>
    );
  }
}

export default Calendar;
