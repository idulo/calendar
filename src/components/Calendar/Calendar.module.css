.container {
  display: flex;
}

.sidepanel {
  display: flex;
  flex-direction: column;
  font-size: large;
  flex-grow: 1;
  margin-left: 3rem;
  margin-top: 3rem;
}

.calendarContainer {
  border: 0.5rem double #7b5e7b;
  border-radius: 1rem;
}

@media only screen and (min-width: 600px) {
  .container {
    margin: 1rem;
  }
}

@media only screen and (min-width: 1000px) {
  .container {
    margin: 3rem 15rem;
  }
}

@media only screen and (min-width: 1400px) {
  .container {
    margin: 3rem 30rem;
  }
}
