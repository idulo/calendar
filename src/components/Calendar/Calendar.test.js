import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import Calendar from "./Calendar";

describe("DaysOfWeek", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<Calendar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly for current month", () => {
    const MonthPicker = create(<Calendar />);

    expect(MonthPicker.toJSON()).toMatchSnapshot();
  });
});
