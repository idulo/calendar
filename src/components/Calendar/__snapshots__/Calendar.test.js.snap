// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaysOfWeek renders correctly for current month 1`] = `
<div
  className="container"
>
  <div
    className="calendarContainer"
  >
    <div
      className="container"
    >
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
        />
      </svg>
      September 2019
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
        />
      </svg>
    </div>
    <div
      className="container"
    >
      <table
        cellPadding="2"
        cellSpacing="2"
        className="calendarDays"
      >
        <thead>
          <tr
            className="container"
          >
            <td>
              Sun
            </td>
            <td>
              Mon
            </td>
            <td>
              Tue
            </td>
            <td>
              Wed
            </td>
            <td>
              Thu
            </td>
            <td>
              Fri
            </td>
            <td>
              Sat
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              1
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              2
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              3
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              4
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              5
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              6
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              7
            </td>
          </tr>
          <tr>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              8
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              9
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              10
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              11
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              12
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              13
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              14
            </td>
          </tr>
          <tr>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              15
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              16
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              17
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              18
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              19
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              20
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              21
            </td>
          </tr>
          <tr>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              22
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              23
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              24
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              25
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              26
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              27
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              28
            </td>
          </tr>
          <tr>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              29
            </td>
            <td
              className="dayCell"
              onClick={[Function]}
            >
              30
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="sidepanel"
  />
</div>
`;
